### PR TITLE
Ability to ignore top bar content inset

### PR DESCRIPTION
**Pull request by Lumenela**
Added method to `HUBViewControllerDelegate` to ask if need to ignore top bar (nav bar + status bar) content inset.

### CODE REVIEW

File: include/HubFramework/HUBViewController.h [hunk 1]
```diff
@@ -152,6 +152,14 @@ NS_ASSUME_NONNULL_BEGIN
  */
 - (BOOL)viewControllerShouldIgnoreHeaderComponentContentInset:(HUBViewController *)viewController;
 
+/**
+ *  Sent to a Hub Framework view controller's delegate to ask if content inset for top bar
+ *  (navigation bar and status bar) should be ignored
+ *
+ *  @param viewController The view controller which displays a header component
+ */
+- (BOOL)viewControllerShouldIgnoreTopBarContentInset:(HUBViewController *)viewController;
```
Comment (JohnSundell) [inline include/HubFramework/HUBViewController.h]: Could this be merged into the API that was previously introduced for header components, and be something like `viewControllerShouldAutomaticallyManageTopContentInset:`?

**Comment by Lumenela** [comment]
@JohnSundell I've replaced to delegate methods for managing content insets with one. Ready for review.

File: tests/HUBViewControllerTests.m [hunk 2]
```diff
@@ -91,7 +91,7 @@ @interface HUBViewControllerTests : XCTestCase <HUBViewControllerDelegate>
 @property (nonatomic, assign) BOOL didReceiveViewControllerDidFinishRendering;
 @property (nonatomic, copy) void (^viewControllerDidFinishRenderingBlock)(void);
 @property (nonatomic, copy) BOOL (^viewControllerShouldStartScrollingBlock)(void);
-@property (nonatomic, copy) BOOL (^viewControllerShouldIgnoreHeaderComponentInset)(void);
+@property (nonatomic, copy) BOOL (^viewControllerShouldAutomaticallyManageTopContentInset)(void);
```
Comment (Goos) [inline tests/HUBViewControllerTests.m]: Just out of curiosity, how come these are blocks rather than just `BOOL` properties?

File: sources/HUBViewController.m [hunk 3]
```diff
@@ -973,21 +973,26 @@ - (nullable HUBComponentWrapper *)componentWrapperFromCell:(HUBComponentCollecti
 
 - (CGFloat)calculateTopContentInset
 {
+    id<HUBViewControllerDelegate> delegate = self.delegate;
+
+    if (![delegate viewControllerShouldAutomaticallyManageTopContentInset:self]) {
```
Comment (JohnSundell) [inline sources/HUBViewController.m]: This means that if the delegate is `nil`, the view controller won't manage content inset, which is not the behavior we want (the default should be to automatically manage). So you'll need to `nil`-check the delegate first.

File: tests/HUBViewControllerTests.m (hunk 2 above)
Comment (JohnSundell) [inline tests/HUBViewControllerTests.m]: I think we started doing it because we needed flexibility of controlling the implementation in some places, and then as multiple devs have added new tests they've stuck to that convention. I'm fine with changing these to simple `BOOL`s, but it should be done in a separate PR that doesn't contain any logic changes, for satefy 😄
